src/components/warnings.ts

```diff
@@ -102,3 +108,30 @@ export function setupWarningsPanel({ element, listeners }: WarningsPanelOptions)
 function formatPosition(start: number, end: number | undefined) {
 	return start === end || !end ? String(start) : [start, end].join('-');
 }
+
+function createSeverity(severity: Warning['severity']) {
+	const el = document.createElement('span');
+
+	el.textContent = severity;
+	el.setAttribute('data-sd-severity', severity);
+
+	return el;
+}
+
+function createInvalidOptionWarnings(optionWarnings: ReadonlyArray<{ text: string }>) {
+	return optionWarnings.map(({ text }) => {
+		const li = document.createElement('li');
+
+		li.appendChild(document.createElement('span')); // dummy
+		li.appendChild(createSeverity('error'));
+
+		const message = document.createElement('span');
+
+		message.textContent = text;
+
+		li.appendChild(message);
+		li.appendChild(document.createElement('span')); // dummy
```

Comment:
Maybe you could use `document.createDocumentFragment` instead?

```suggestion
		const fragment = document.createDocumentFragment();
		const message = document.createElement('span');
		const = createSeverity('error');
		const dummy = () => document.createElement('span');

		message.textContent = text;
		[dummy(), severity, message, dummy()].forEach(e => fragment.appendChild(e));
		li.appendChild(fragment);
```

Reply:
Maybe we could do it, but I want to keep it because the current code around `result.warnings` doesn't use `createDocumentFragment`. It would be better to refactor them together with another PR.